webservices/rest/src/main/java/com/epam/rest/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.rest.dao;
+
+
+import com.epam.rest.domain.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    private final Map<Integer, User> storage = new HashMap<>();
+    private int counter = 0;
```

Comment:
Thanks, forgot about it.
As I see after a bit of googling it is a matter of a trade-off between consistency and performance. But as you asked about the minimum performance impact, then ConcurrentHashMap is preferable.
Replaced.